File: python/dictation.py
from soniox.transcribe_file import transcribe_file_short
from soniox.speech_service import SpeechClient


def main():
    with SpeechClient() as client:
        result = transcribe_file_short(
            "../test_data/test_audio_dictation.mp3", client, enable_dictation=True
        )
        print(" ".join(w.text for w in result.words))


if __name__ == "__main__":
    main()

File: node/dictation.js
const { SpeechClient } = require("@soniox/soniox-node");

/**
 * Set your Soniox Cloud API key:
 * from command line: export SONIOX_API_KEY=<YOUR-API-KEY>
 * or
 * pass config object: new SpeechClient({ api_key: "<YOUR-API-KEY>" })
 */
const speechClient = new SpeechClient();

(async function () {
  const result = await speechClient.transcribeFileShort(
    "../test_data/test_audio_dictation.mp3",
    {
      enable_dictation: true,
    }
  );

  console.log(`Words: ${result.words.map((word) => word.text).join(" ")}`);
})();
